src/test/kotlin/lotto/LottoShopTest.kt

```diff
@@ -0,0 +1,50 @@
+package lotto
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class LottoShopTest {
+
+    @Test
+    fun `원하는 개수만큼의 티켓을 발급하는지 확인`() {
+        // given
+        val ticketCount = 10
+        val lottoShop = LottoShop(LottoTicketGenerator())
+
+        // then
+        assertThat(lottoShop.getAutoTickets(10_000)).hasSize(ticketCount)
+    }
+
+    @Test
+    fun `행운 숫자와 비교해서 모든 로또 결과 구하기`() {
+        // given
+        val lottoNumbers = listOf(
+            LottoNumber(1),
+            LottoNumber(2),
+            LottoNumber(3),
+            LottoNumber(4),
+            LottoNumber(5),
+            LottoNumber(6)
+        )
+        val lottoTicket = LottoTicket(lottoNumbers)
```

Comment:
생성자의 수가 늘어날수록 클래스를 사용하기 위한 유연성은 높아집니다.

```suggestion
        val lottoTicket = LottoTicket(1, 2, 3, 4, 5, 6)
```